oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeState.java

```diff
@@ -414,6 +414,11 @@ public String toString() {
      * @param revision the revision this node is created.
      */
     UpdateOp asOperation(@NotNull Revision revision) {
+
+        if (Utils.isNodeNameLong(path, store.getDocumentStore().getNodeNameLimit())) {
+            throw new DocumentStoreException("Node name is too long: " + path);
+        }
+
```

Comment:
Good point, `UpdateOp` doesnt have the `path` .. and `asOperation` is used as part of `addNode`, so it does make sense. I'm just wondering if that covers 100% of the cases.. hmm

Reply:
What I was wondering, what about moving this check from the somewhat implicit `asOperaiton` location to the a bit more explicit `CommitBuilder.addNode(Path)`? It should have the same effect, as `addPath` does eventually call `asOperation` .. but having it in `addNode(Path)` would seem a bit more logical for people reading the code.

But the question would still be somewhat open: is this the only place where long node names can be added .. maybe @mreutegg / @reschke can comment too. The only suspicious place I found was `SplitOperations.createIntermediateDocs` - but that only adds a few characters to the id and therefore is probably fine.

Reply:
@stefan-egli , do you mean these lines?

https://github.com/apache/jackrabbit-oak/blob/c6ddcc55bee3de915459af01e91edad32d538f3d/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/SplitOperations.java#L317-L318

That should be fine because the ID of an intermediate document always ends with a sequence of digits that indicate the height of the split document tree. This is usually just one digit.

Reply:
yes those lines.

Ok, I'll mark this as resolved. I've double checked again and found no other cases calling `isLongPath` (that would now get the name too-long check removed).